packages/davinci-client/src/lib/authorize.utils.ts

```diff
@@ -37,6 +37,9 @@ export async function createAuthorizeUrl(
   const [authorizeUrlOptions, storeOptions] = generateAndStoreAuthUrlValues({
     clientId: options.clientId,
     login: options.login,
+    // this type fails when module resolution is Node16
+    // Probably because the javascript-sdk is set to bundler
+    // so we need to make this correct with .js extensions if possible
     serverConfig: { baseUrl },
```

Comment:
This type fails when `moduleResolution` is `NodeNext` or `Node16`.

This is because the legacy SDK uses `bundler` meaning the "host" application (where this is consumed) is responsible for the import/exports. But because this package is using NodeNext, it doesn't know how to resolve the imports from the SDK.

This causes the type failure.

Reply:
The reason here - is actually (seemingly) specifically because the d.ts files _do not have extensions_ but the outputted JS of the legacy SDK does have extensions.

This would be how to resolve this without changing the module_resolution of the legacy SDK ( I think ).

Reply:
So, if it flat out fails, how do we merge this PR?

Reply:
it works - because i made the moduleResolution "Bundler" which is how its working currently in `main`.

But its just a note that when we have no legacy SDK dependency, we should move to Node16/Node18/NodeNext (one of them).

Or agree that "we only target bundlers". So most likely the former.